File: Python/Data Structures & Algorithms Module - Python/Revision/BinarySearchTree/BSTApp.py
"""
BSTApp.py
Billy Cussen
18/01/2021
"""

from BST import BST

myBST = BST()

print("Is Empty: "+str(myBST.isEmpty()))
print("Size: "+str(myBST.size()))
print("Depth: "+str(myBST.depth()))
myBST.insert(20)
myBST.insert(6)
myBST.insert(26)
myBST.insert(5)
myBST.insert(7)
myBST.insert(21)
myBST.insert(30)
myBST.inOrder()
print("Is Empty: "+str(myBST.isEmpty()))
print("Size: "+str(myBST.size()))
print("Depth: "+str(myBST.depth()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))

File: Java/Data Structures & Algorithms Module - Java/Revision/BinarySearchTree/BSTApp.java
/*
BSTApp.java
Billy Cussen
18/01/2021
*/
public class BSTApp {
    public static void main(String[]args){

        BST<Integer> myBST = new BST<Integer>();
        System.out.println("Size: "+myBST.size());
        System.out.println("Is Empty: "+myBST.isEmpty());
        System.out.println("Depth: "+myBST.depth());
        myBST.insert(20);
        myBST.insert(12);
        myBST.insert(5);
        myBST.insert(14);
        myBST.insert(30);
        myBST.insert(25);
        myBST.insert(35);
        System.out.println("Size: "+myBST.size());
        System.out.println("Is Empty: "+myBST.isEmpty());
        System.out.println("Depth: "+myBST.depth());
        System.out.println("Max: "+myBST.max());
        System.out.println("Min: "+myBST.min());
        System.out.println(myBST.inOrder());
    }

}
